### ISSUE TITLE
jsonschema.exceptions.ValidationError on schema['properties']['billing_address']['properties']['address2']

### ISSUE THREAD

**Issue by farzeni** [Matatika/tap-shopify#8]
Hi, 

I'm trying to run this tap with meltano. I have a problem with abandoned_checkouts and customers collections on address2 property.

It looks like the tap expects the field to be a string but in my case it contains a null value. 

here part of the error 

jsonschema.exceptions.ValidationError: None is not of type 'string'
Failed validating 'type' in schema['properties']['billing_address']['properties']['address2']
{'type': 'string'}
On instance['billing_address']['address2']
None

Do you know if I can do something on my side to avoid this, like adding some transformations, or it is a problem that should be address directly in the tap?

Thanks

**Comment by farzeni** [comment]
Thank you for your quick response. 

I appreciate your suggestions and will try using a custom catalog and stream maps to handle this issue.

Looking forward to the permanent fix 

Thanks again

**Comment by mike-luabase** [comment]
@farzeni @ReubenFrankel did you end up fixing this with a stream map? It wasn't clear to me how to map nested field. I was trying:

```
plugins:
  extractors:
  - name: tap-shopify
    variant: matatika
    pip_url: git+https://github.com/Matatika/tap-shopify.git
    config:
      stream_maps:
        abandoned_checkouts:
          phone: __NULL__
```

but getting:

```
jsonschema.exceptions.ValidationError: None is not of type 'string' cmd_type=elb consumer=True name=target-snowflake producer=False stdio=stderr string_id=target-snowflake
2023-08-24T02:16:15.463820Z [info     ]                                cmd_type=elb consumer=True name=target-snowflake producer=False stdio=stderr string_id=target-snowflake
2023-08-24T02:16:15.463907Z [info     ] Failed validating 'type' in schema['properties']['billing_address']['properties']['phone']: cmd_type=elb consumer=True name=target-snowflake producer=False stdio=stderr string_id=target-snowflake
2023-08-24T02:16:15.463999Z [info     ]     {'type': 'string'}         cmd_type=elb consumer=True name=target-snowflake producer=False stdio=stderr string_id=target-snowflake
2023-08-24T02:16:15.464083Z [info     ]                                cmd_type=elb consumer=True name=target-snowflake producer=False stdio=stderr string_id=target-snowflake
2023-08-24T02:16:15.464169Z [info     ] On instance['billing_address']['phone']: cmd_type=elb consumer=True name=target-snowflake producer=False stdio=stderr string_id=target-snowflake
2023-08-24T02:16:15.464293Z [info     ]     None 
```

**Comment by farzeni** [comment]
Hi @mike-luabase ,

In the end, I have opted to use a custom catalog because stream maps have limitations with nested fields:
https://sdk.meltano.com/en/latest/stream_maps.html#no-nested-property-declarations-or-removals

I've added a modified [tap-shopify.catalog.json](https://gist.github.com/farzeni/49fb30e6506453fced4f71043e62bafd) file under the extract folder and set it up in the plugin configuration.

```
plugins:
  extractors:
    - name: tap-shopify
      variant: matatika
      catalog: extract/tap-shopify.catalog.json
      pip_url: git+https://github.com/Matatika/tap-shopify.git
      config:
        store: ....
        start_date: ...
        is_plus_account: false
```

**Comment by ReubenFrankel** [comment]
Various schema issues addressed in [`v0.2.0`](https://github.com/Matatika/tap-shopify/releases/tag/v0.2.0)
